source/solvers/navier_stokes_vof_assemblers.cc

```diff
@@ -569,6 +569,67 @@ GLSNavierStokesVOFAssemblerMarangoni<dim>::assemble_rhs(
 template class GLSNavierStokesVOFAssemblerMarangoni<2>;
 template class GLSNavierStokesVOFAssemblerMarangoni<3>;
 
+template <int dim>
+void
+NavierStokesVOFAssemblerEvaporation<dim>::assemble_matrix(
+  NavierStokesScratchData<dim> & /*scratch_data*/,
+  StabilizedMethodsTensorCopyData<dim> & /*copy_data*/)
+{}
+
+template <int dim>
+void
+NavierStokesVOFAssemblerEvaporation<dim>::assemble_rhs(
+  NavierStokesScratchData<dim>         &scratch_data,
+  StabilizedMethodsTensorCopyData<dim> &copy_data)
+{
+  // Loop and quadrature information
+  const auto        &JxW        = scratch_data.JxW;
+  const unsigned int n_q_points = scratch_data.n_q_points;
+  const unsigned int n_dofs     = scratch_data.n_dofs;
+
+  // Copy data elements
+  auto &strong_residual = copy_data.strong_residual;
+  auto &local_rhs       = copy_data.local_rhs;
+
+  // Loop over the quadrature points
+  for (unsigned int q = 0; q < n_q_points; ++q)
+    {
+      // Only temporary, should change if the evaporation also depends on the
+      // species concentration
+      std::map<field, double> field_value;
+      if (this->evaporation_model->depends_on(field::temperature))
+        {
+          field_value[field::temperature] = scratch_data.temperature_values[q];
+        }
+
+      // Recoil pressure
+      const double recoil_pressure =
+        this->evaporation_model->momentum_flux(field_value);
+
+      // Gather phase fraction gradient
+      const Tensor<1, dim> &phase_gradient_value =
```

Comment:
Same thing here, I would specify here that it is the filtered one `filtered_phase_gradient_q`